pilot/pkg/model/jwks_resolver.go

```diff
@@ -289,6 +289,12 @@ func (r *JwksResolver) getRemoteContentWithRetry(uri string, retry int) ([]byte,
 		return nil, err
 	}
 
+	// Default to https if the provided url has no scheme
```

Comment:
I think the proper place to do this should be in `SetAuthenticationPolicyJwksURIs()`, and that would allow you to write the unit test easily in `TestSetAuthenticationPolicyJwksURIs()`.

Reply:
I don't think `SetAuthenticationPolicyJwksURIs` is the best place - it loops over two structures calling `resolveJwksURIUsingOpenID` for each - we'd need to fix this in both places.

I'm trying to put it in `resolveJwksURIUsingOpenID`, and instead of using native url parsing (which doesn't handle the mock server urls without a scheme), do something just with string parsing

Reply:
Ok, how's that?